src/NUnitFramework/framework/Internal/RandomGenerator.cs

```diff
@@ -180,5 +181,55 @@ public T GetEnum<T>()
         }
 
         #endregion
+        
+        #region String & Guids
+        
+        /// <summary>
+        /// Default characters for random functions.
+        /// </summary>
+        /// <remarks>Default characters are the English alphabet (uppercase &amp; lowercase), arabic numerals, and underscore</remarks>
+        public const string DefaultStringChars = "abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNOPQRSTUVWXYZ0123456789_";        
+        
+        /// <summary>
+        /// Generate a random Guid
+        /// </summary>
+        /// <returns>string GUID</returns>
+        public string CreateRandomGuid()
+        {
+            return Guid.NewGuid().ToString();
+        }
+        
+        /// <summary>
+        /// Generate a random string based on the characters from the input string.
+        /// </summary>
+        /// <param name="outputLength">desired length of output string.</param>
+        /// <param name="allowedChars">string representing the set of characters from which to construct the resulting string</param>
+        /// <returns>A random string of arbitrary length</returns>
+        public string CreateRandomString(int outputLength, string allowedChars)
```

Comment:
As with Guid, best named like the other methods, which are all called like
    TestContext.CurrentContext.Random.GetXxxx